plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/conditions/NestedConditionEvaluator.java

```diff
@@ -0,0 +1,129 @@
+        }
+
+
+        try {
+            // Get list of nested items to be evaluated
+            Object nestedItems = propertyConditionEvaluator.getPropertyValue(item, path);
+            if (nestedItems instanceof List) {
+
+                // Evaluated each nested items until one match the nested condition
+                for (Object nestedItem : (List<Object>) nestedItems) {
+                    if (nestedItem instanceof Map) {
+                        Map<String, Object> flattenedNestedItem = flattenNestedItem(path, (Map<String, Object>) nestedItem);
+                        Item finalNestedItem = createFinalNestedItemForEvaluation(item, path, flattenedNestedItem);
+                        if (finalNestedItem != null && dispatcher.eval(subCondition, finalNestedItem, context)) {
+                            // We found at least one nested item matching
+                            return true;
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Failed to evaluated nested condition", e);
+            return false;
+        }
+        return false;
+    }
+
+    protected Map<String, Object> flattenNestedItem(String path, Map<String, Object> nestedItem) {
+        Map<String, Object> flattenNestedItem = new HashMap<>();
+
+        // Merge the nested Item in flat properties
+        if (StringUtils.isNotEmpty(path)) {
+            // substring to keep only the part after "properties".
+            // For example in case of "properties.interests", we only want to keep "interests"
+            String propertyPath = StringUtils.substringAfter(path, ".");
+            if (StringUtils.isNotEmpty(propertyPath)) {
+                String[] propertyKeys = propertyPath.split("\\.");
+                Iterator<String> propertyKeysIterator = Arrays.stream(propertyKeys).iterator();
+
+                Map<String, Object> currentPropertiesLevel = flattenNestedItem;
+                while (propertyKeysIterator.hasNext()) {
+                    String propertyKey = propertyKeysIterator.next();
+                    if (!propertyKeysIterator.hasNext()) {
+                        // last property, we set the nested Item
+                        currentPropertiesLevel.put(propertyKey, nestedItem);
+                    } else {
+                        // new level of prop
+                        Map<String, Object> subLevel = new HashMap<>();
+                        currentPropertiesLevel.put(propertyKey, subLevel);
+                        currentPropertiesLevel = subLevel;
+                    }
+                }
+            }
+        }
+
+        return flattenNestedItem;
+    }
+
+    protected Item createFinalNestedItemForEvaluation(Item parentItem, String path, Map<String, Object> flattenedNestedItem) {
+        // Build a basic Item with merged properties
+        if (parentItem instanceof Profile) {
```

Comment:
What about the other kind of item (like customItems) ?

Reply:
they cannot provide ES Mapping anyway, so they cannot use nested ES mapping. 
So for now I would say it's not in the current scope.